src/resources/elements/tokenSelect/tokenSelect.ts

```diff
@@ -30,6 +30,7 @@ export class TokenSelect {
 
   async attached(): Promise<void> {
     if (!this.tokenInfos) {
+      this.tokenInfos = Array(this.tokenAddresses.length).fill({} as ITokenInfo);
```

Comment:
since we are waiting for the entire list, not for individual items, I might have put the spinner in the placeholder instead of every item.  Would that make sense?  Could be simpler to code and better UX.

Reply:
I've simplified the code. commit a19ce14